tests/tasks/test_tasks.py

```diff
@@ -43,8 +43,8 @@ def wait_for_task_completion():
         while task_state in WAIT_STATUSES:
             if elapsed_seconds == TIME_LIMIT:
```

Comment:
I think it'd be better to change `==` to `>=` here. If for some reason you change `TIME_LIMIT` to a non-integer number, you may end up in an infinite loop with the current condition.